dags/burnham.py

```diff
@@ -0,0 +1,143 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import datetime
+import json
+import uuid
+
+from airflow import models
+from utils.burnham import (
+    new_burnham_bigquery_operator,
+    new_burnham_operator,
+    new_burnham_sensor,
+)
+
+DAG_OWNER = "rpierzina@mozilla.com"
+DAG_EMAIL = ["telemetry-alerts@mozilla.com", "rpierzina@mozilla.com"]
+
+QUERY_TEMPLATE = """
+SELECT
+  technology_space_travel.key,
+  SUM(technology_space_travel.value) AS value_sum
+FROM
+  `{project_id}.burnham_live.discovery_v1`
+CROSS JOIN
+  UNNEST(metrics.labeled_counter.technology_space_travel) AS technology_space_travel
+WHERE
+  submission_timestamp > TIMESTAMP_SUB(CURRENT_TIMESTAMP(), INTERVAL 3 HOUR)
+  AND metrics.uuid.test_run = "{test_run}"
+  AND metrics.string.test_name = "{test_name}"
+GROUP BY
+  technology_space_travel.key
+ORDER BY
+  technology_space_travel.key
+LIMIT
+  {limit}
+"""
+
+SENSOR_TEMPLATE = """
+SELECT
+  COUNT(*) >= 10
+FROM
+  `{project_id}.burnham_live.discovery_v1`
+WHERE
+  submission_timestamp > TIMESTAMP_SUB(CURRENT_TIMESTAMP(), INTERVAL 3 HOUR)
+  AND metrics.uuid.test_run = "{test_run}"
+  AND metrics.string.test_name = "{test_name}"
+"""
+
+default_args = {
+    "owner": DAG_OWNER,
+    "email": DAG_EMAIL,
+    "email_on_failure": True,
+    "email_on_retry": True,
+    "start_date": datetime.datetime(2020, 6, 15),
+    "depends_on_past": False,
+    "retries": 0,
+}
+
+with models.DAG(
+    "burnham", schedule_interval="@daily", default_args=default_args,
+) as dag:
+
+    # Generate a UUID for this test run
+    burnham_test_run = str(uuid.uuid4())
```

Comment:
Generating a UUID here is unlikely to do what you might expect. See https://stackoverflow.com/questions/55748050/generating-uuid-and-use-it-across-airflow-dag

The top answer there has an example of using XCOM that happens to be exactly what we're here: calling `uuid.uuid4()`. I think we should follow exactly what they do there:

```suggestion
    generate_burnham_test_run_uuid = PythonOperator(
        task_id='generate_burnham_test_run_uuid',
        python_callable=lambda: str(uuid.uuid4())
    )
    burnham_test_run = '{{ task_instance.xcom_pull("generate_uuid") }}'
```

Note that the contents of `burnham_test_run` will literally be `'{{ task_instance.xcom_pull("generate_uuid") }}'`. The contents only get replaced when the individual operators run. Airflow operators can designate different parameters as "templated" meaning that the strings passed in will have variables replaced before running the body of the operator. It looks like all the relevant places this ID gets used _are_ templated params, so I think we're good.

Reply:
See for example https://airflow.apache.org/docs/stable/_api/airflow/contrib/operators/kubernetes_pod_operator/index.html#airflow.contrib.operators.kubernetes_pod_operator.KubernetesPodOperator for docs that show which params are templated.

Reply:
Thank you! I updated the UUID generation as you suggested. Note that I used the task ID in the template.